### PR TITLE
Allow regexps for fieldnames in patterns

### LINKED ISSUE #512: regex not working in JS object key
https://sgrep.live/4Xz does not work but 

if it's a hard coded value it does: https://sgrep.live/PvW

### PR DESCRIPTION
**Pull request by aryx**
This closes #512

Test plan:
test file included

### CODE REVIEW

**Review by DrewDennison — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,18 @@
 
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## unrelease
+
+### Added
+- You can now use regexps for field names:
+  ```
+  var $X = {"=~/[lL]ocation/": $Y};
```
Comment (brendongo) [inline CHANGELOG.md:10]: Is there a reason we went with `=~` as syntax to use regex?

Comment (DrewDennison) [reply CHANGELOG.md:10]: I'm guessing an very old design choice. All other regexes work like this https://docs.semgrep.dev/config/simple#string-matching so this is more of a bugfix PR to allow it on keys

Comment (aryx) [reply CHANGELOG.md:10]: it comes from Perl, which uses this =~ operator. Perl is the regexp language :)

Comment (aryx) [reply CHANGELOG.md:10]: what do you propose?

**Comment by DrewDennison** [comment]
@ajinabraham thanks again for the bug report!

File: CHANGELOG.md (hunk 1 above)
Comment (DrewDennison) [reply CHANGELOG.md:10]: https://xkcd.com/208/

Comment (brendongo) [reply CHANGELOG.md:10]: Just wondering. It's not the best from the standpoint of a new user coming in but if it exists elsewhere then it's easier to justify

**Review by brendongo — APPROVED** [review]

**Pull request merged**